boards/DIM/Src/App/App_ErrorTable.c

```diff
@@ -0,0 +1,216 @@
+#include <stdlib.h>
+#include <stdbool.h>
+#include <assert.h>
+#include "App_ErrorTable.h"
+
+// clang-format off
+struct ErrorTable
+{
+    bool bms_stack_watermark_above_threshold_task1hz;
+    bool dcm_stack_watermark_above_threshold_task1hz;
+    bool fsm_stack_watermark_above_threshold_task1hz;
+    bool pdm_stack_watermark_above_threshold_task1hz;
+    bool dim_stack_watermark_above_threshold_task1hz;
+    bool bms_dummy_critical_error;
+    bool dcm_dummy_critical_error;
+    bool fsm_dummy_critical_error;
+    bool pdm_dummy_critical_error;
+    bool dim_dummy_critical_error;
+};
+
+struct ErrorTable* App_ErrorTable_Create(void)
+{
+    struct ErrorTable* error_table = malloc(sizeof(struct ErrorTable));
+    assert(error_table != NULL);
+
+    return error_table;
+}
+
+void App_ErrorTable_Destroy(struct ErrorTable* error_table)
+{
+    free(error_table);
+}
+
+uint32_t App_ErrorTable_HasNonCriticalError(struct ErrorTable* error_table, struct Error errors[])
```

Comment:
Um. I'm a tad confused. This is a function in the `DIM` board (it should be shared?) and it also uses the stack watermark from the BMS for some reason?

Reply:
It was just placed here as an example so I can check that the code compiles. And the stack watermark is just one of the example non-critical error in the error table.